SitecorePackageDeployer/Tasks/InstallPackage.cs

```diff
@@ -210,6 +224,9 @@ private void InstallPackages()
                             installationHistoryRoot = ex.HistoryPath;
                             installLogger.Fatal("Package install failed", ex);
 
+                            package.Status = SitecorePackageStatus.Failed;
```

Comment:
Status is updated to "Failed" and added to the List, then the exception is thrown, and caught in the next block- which will then add the failure again- on line 257.